x-pack/plugins/ml/public/application/routing/routes/settings/settings.tsx

```diff
@@ -42,11 +42,18 @@ const PageWrapper: FC<PageProps> = ({ deps }) => {
   useTimefilter({ timeRangeSelector: false, autoRefreshSelector: false });
 
   const canGetFilters = checkPermission('canGetFilters');
+  const canCreateFilter = checkPermission('canCreateFilter');
   const canGetCalendars = checkPermission('canGetCalendars');
+  const canCreateCalendar = checkPermission('canCreateCalendar');
 
   return (
     <PageLoader context={context}>
-      <Settings canGetCalendars={canGetCalendars} canGetFilters={canGetFilters} />
+      <Settings
+        canGetCalendars={canGetCalendars}
+        canCreateFilter={canCreateFilter}
+        canGetFilters={canGetFilters}
+        canCreateCalendar={canCreateCalendar}
```

Comment:
nit: I see you propagate these props to another child component inside of `Settitngs`, so it might worth to create a context for settings

Reply:
Done in https://github.com/elastic/kibana/pull/68663/commits/e67c1fb40fd8180d877148fc18d379610b21e5d6